### PR TITLE
Artifactory+ptf

### PR DESCRIPTION
**Pull request by John-A-Davies**
Please check if your PR fulfills the following requirements. This is simply a reminder of what we are going to look for before merging your PR. If you don't know all of this information when you create this PR, don't worry. You can edit this template as you're working on it.

- [y] Tests for the changes have been added (for bug fixes / features)
- [n] Necessary documentation (if appropriate) have been added / updated
- [y] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix 
- [x] Feature  Also publish PTF
- [ ] Other... Please describe:

#### Relevant issues

Fixes 

#### Changes proposed in this PR

- Adds the PTF to the list of files pushed to Artifactory

#### Does this PR introduce a breaking change?

- [ ] Yes
- [n] No

#### Does this PR do something the person installing Zowe should know about?

---
* Affected function: _general area of interest_ *

---
* Description: _1 line description_ *

---
* Part: _name of customizable file involved_  *

---
_multi-line description_

#### Is there a related doc issue or Pull Request? 

Doc issue/PR number:

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -147,7 +147,8 @@ sed -e 's#{BUILD_BRANCH}#${env.BRANCH_NAME}#g' \
       '.pax/zowe-smpe.zip',
       '.pax/smpe-promote.tar',
       '.pax/pd.htm',
-      '.pax/smpe-build-logs.pax.Z'
+      '.pax/smpe-build-logs.pax.Z',
+      '.pax/AZWE001.TMP0001.zip'
```
Comment (OnnoVdT) [inline Jenkinsfile]: We are missing .pax/AZWE001.zip. As soon as Nayer pushes the ptf-bucket into the staging branch, zowe-smpe.zip will no longer hold the FMID but the PTF. (Note, zowe-smpe.zip is intended to hold what should be published on zowe.org.)  Both FMID and PTF should always be available in Artifactory for automated testing. (Also, the FMID is needed to simplify loading a PTF build in Shopz.)

**Comment by OnnoVdT** [comment]
@jackjia-ibm 
if the FMID built by the branch is present in Artifactory, it can be used to drastically simplify redistribution of the PTF (which has the same code) via Shopz, so I'm in favor of leaving it in. 

For the moment we can indeed drop the need to also place ptf.zip in Artifactory, but it comes with some complications. We do have to think about when we start building our second FMID. If we want to keep automated test for applying service active in that stream (which I think we would) then ptf.zip should be back in Artifactory (in that stream zowe-smpe.zip holds the FMID). 

And to keep test (and up to a point Shopz) automation simple, it would be best if it can always use fixed names for the input files, regardless the build running in GA or PTF mode, so no reliance on zowe-smpe.zip, but only on fmid.zip and ptf.zip. zowe-smpe.zip is basically meant for zowe.org.

**Comment by jackjia-ibm** [comment]
@OnnoVdT I think I'm not convinced we publish the fmid built from branch to Artifactory yet. I put the formal FMID 1.9.0 onto Marist. If you use Marist for testing, we can assume the FMID is always there:

```
ZOWEAD3:/ZOWE/zowe-fmids: >ls -la
total 959872
drwxr-xr-x   2 ZOWEAD3  IPGROUP     8192 Mar  5 08:08 .
drwxrwxrwx+ 17 OMVSKERN IPGROUP     8192 Mar  5 08:15 ..
-rw-r--r--   1 ZOWEAD3  IPGROUP    64476 Mar  5 08:08 AZWE001.htm
-rw-r--r--   1 ZOWEAD3  IPGROUP  491097600 Mar  5 08:13 AZWE001.pax.Z
-rw-r--r--   1 ZOWEAD3  IPGROUP     9412 Mar  5 08:08 AZWE001.readme.txt
```

In the future if we have new formal FMID, like AZWE002, we can also put here. This should reduce unnecessary transfer of FMID. We can configure this as a test config item for marist server, where to get the Zowe FMID to test PTF.

And for future releases, if we built v1.x PTFs and v2.x FMID at same time, we will have `v1.x/staging` to build PTFs and the current `staging` will be reset to build FMID instead of PTF.  This branch design has been defined for a while, please let me know if it doesn't match the SMPE scenario.

Please let me know I still missed some points here. Thanks!

For testing a SMPE build (FMID/PTF), the team points to exact build URL, like they point to test `libs-snapshot-local/org/zowe/1.9.0-SNAPSHOT/zowe-smpe-1.9.0-snapshot-803-20200228191618.zip`, so I don't think filename really matters here.

**Comment by OnnoVdT** [comment]
The IBM redistribution of Zowe FMID and PTF relies on the FMID being available in Artifactory with the current build in it. For FMID redistribution this is obvious, and it can be obtained from zowe-smpe.zip, but it is also used for PTF redistribution. Here zowe-smpe.zip holds the PTF, not the FMID, so fmid.zip must be added explicitly to Artifactory. 
The reason we use the FMID is that its format is easily consumable. The only thing that can process a PTF is SMPE, and that only during APPLY, which means we must first install the Zowe FMID, then apply (and accept) the PTF on top of it. After accept of the PTF, we have consumable data in the distribution libraries. Data that matches what is already in fmid.zip. 
Adding install of FMID and PTF to the redistribution process introduces a bunch of places where things can go wrong, or slight changes could get introduced, which are things I'm really trying to avoid. By using the fmid.zip package all we do is pick up the data as produced by the open source community, which is a straightforward and clean process.

**Comment by jackjia-ibm** [comment]
Thanks @OnnoVdT @John-A-Davies for the explanation. The changes are made and here are the artifacts published:

```
artifactoryUpload; see: https://jenkins.io/redirect/pipeline-cps-method-mismatches/
[consumer_0] Deploying artifact: https://zowe.jfrog.io/zowe/libs-snapshot-local/org/zowe/1.9.0-PR-1180/zowe-1.9.0-pr-1180-6-20200311162738.pax
[consumer_1] Deploying artifact: https://zowe.jfrog.io/zowe/libs-snapshot-local/org/zowe/1.9.0-PR-1180/zowe-smpe-1.9.0-pr-1180-6-20200311162738.zip
[consumer_2] Deploying artifact: https://zowe.jfrog.io/zowe/libs-snapshot-local/org/zowe/1.9.0-PR-1180/smpe-promote-1.9.0-pr-1180-6-20200311162738.tar
[consumer_2] Deploying artifact: https://zowe.jfrog.io/zowe/libs-snapshot-local/org/zowe/1.9.0-PR-1180/pd-1.9.0-pr-1180-6-20200311162738.htm
[consumer_2] Deploying artifact: https://zowe.jfrog.io/zowe/libs-snapshot-local/org/zowe/1.9.0-PR-1180/smpe-build-logs-1.9.0-pr-1180-6-20200311162738.pax.Z
[consumer_2] Deploying artifact: https://zowe.jfrog.io/zowe/libs-snapshot-local/org/zowe/1.9.0-PR-1180/AZWE001-1.9.0-pr-1180-6-20200311162738.zip
```

**Comment by John-A-Davies** [comment]
zowe-smpe-1.9.0-pr-1180-6-20200311162738.zip
and
AZWE001-1.9.0-pr-1180-6-20200311162738.zip
contain the same data.  They are both FMID.  I think one of them should be the PTF.

**Comment by jackjia-ibm** [comment]
@OnnoVdT could you please verify if the changes in this PR meets what we have discussed?

**Pull request merged**